test/utilities/git.jl

```diff
@@ -1,11 +1,300 @@
+                @test occursin(pushed_str, output)
+            end
+        end
+    end
+
+    @testset "force push" begin
+        mktempdir() do local_remote_dir
+            mktempdir() do f
+                local_remote_path = create_local_remote(local_remote_dir)
+
+                cd(f)
+                run(`git init`)
+                run(`git remote add origin $local_remote_path`)
+
+                run(`touch foobar.txt`)
+                CompatHelper.git_add()
+                CompatHelper.git_commit("Message 1")
+
+                output = read(`git log --decorate`, String)
+                @test !occursin(pushed_str, output)
+
+                CompatHelper.git_push("origin", "master")
+
+                output = read(`git log --decorate`, String)
+                @test occursin(pushed_str, output)
+                @test occursin("Message 1", output)
+
+                run(`touch baz.txt`)
+                CompatHelper.git_add()
+                run(`git -c user.name=user -c user.email=email commit --amend --no-edit -m "Message 2"`)
+
+                CompatHelper.git_push("origin", "master"; force=true)
+                output = read(`git log --decorate`, String)
+                @test occursin(pushed_str, output)
+                @test occursin("Message 2", output)
+
+            end
+        end
+    end
+
+    @testset "SSH push" begin
+        mktempdir() do local_remote_dir
+            mktempdir() do f
+                local_path = joinpath(f, CompatHelper.LOCAL_REPO_NAME)
+                pkey = joinpath(f, "privatekey")
+                open(pkey, "w") do io
+                    print(io, TEST_PKEY)
+                end
+
+                local_remote_path = create_local_remote(local_remote_dir)
+
+                cd(f)
+                run(`git init`)
+                run(`git remote add origin $local_remote_path`)
+
+                run(`touch foobar.txt`)
+                CompatHelper.git_add()
+                CompatHelper.git_commit("Message")
+
+                output = read(`git log --decorate`, String)
+                @test !occursin(pushed_str, output)
+
+                CompatHelper.git_push("origin", "master", pkey)
+
+                output = read(`git log --decorate`, String)
+                @test occursin(pushed_str, output)
+            end
+        end
+    end
+end
+
+@testset "git_commit" begin
+    @testset "success" begin
+        mktempdir() do f
+            cd(f)
+            run(`git init`)
+            run(`touch foobar.txt`)
+            CompatHelper.git_add()
+            @test CompatHelper.git_commit("Message")
+        end
+    end
+
+    @testset "failure" begin
+        mktempdir() do f
+            cd(f)
+            run(`git init`)
+            run(`touch foobar.txt`)
+            CompatHelper.git_add()
+            # Manually create an index lock file, so that the commit fails
+            run(`touch .git/index.lock`)
+            @test !CompatHelper.git_commit("Message")
```

Comment:
```suggestion
            CompatHelper.git_add()

            # Manually create an index lock file, so that the commit fails
            run(`touch .git/index.lock`)

            @test !CompatHelper.git_commit("Message")
```